apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs

```diff
@@ -119,11 +120,16 @@ public string AutoGeneratedCommandTable
         /// <param name="connection">A connection to the Spanner database. Must not be null.</param>
         /// <param name="autoGeneratedCommandTable">The Spanner database table to use for automatically generated commands.
         /// May be null.</param>
-        public SpannerDataAdapter(SpannerConnection connection, string autoGeneratedCommandTable)
+        /// <param name="primaryKeys">The set of columns that form the primary key for <paramref name="autoGeneratedCommandTable"/>.</param>
+        public SpannerDataAdapter(SpannerConnection connection, string autoGeneratedCommandTable, params string[] primaryKeys)
         {
             GaxPreconditions.CheckNotNull(connection, nameof(connection));
             SpannerConnection = connection;
             AutoGeneratedCommandTable = autoGeneratedCommandTable;
+            if (primaryKeys != null)
+            {
+                AutoGeneratedCommandPrimaryKeys.UnionWith(primaryKeys);
+            }
```

Comment:
It is just a matter of preference, but I tend to think lines like this look better with null coalescing.
```
AutoGeneratedCommandPrimaryKeys.UnionWith(primaryKeys ?? Enumerable.Empty<string>());
```